<?xml version="1.0" encoding="utf-8"?>
<launch>
 <!-- <node type="rviz" name="rviz" pkg="rviz" args="-d $(find package_name)/rviz/multi_teleop.rviz" /> -->
 <arg default="car1" name="car1_name"/>
 <arg default="car2" name="car2_name"/>
 <arg default="car3" name="car3_name"/>
 <arg default="car4" name="car4_name"/>
 <arg default="clcbs_data_9" name="bag_name"/>
 <arg default="clcbs_data_test_100" name="record_name"/>
 <!-- Set to 1 if you want to run the map_server -->
 <node args="-q -a -o $(find nhttc_ros)/bags/output_bags/$(arg record_name).bag" name="rosbag_record" output="screen" pkg="rosbag" type="record"/>
 <arg name="map_server" value="1"/>
 <!-- Launch  map server-->
 <group if="$(arg map_server)">
  <include file="$(find mushr_base)/launch/includes/map_server.launch"/>
 </group>
 <!-- Launch first car -->
 <group ns="$(arg car1_name)">
  <include file="$(find mushr_sim)/launch/single_car.launch">
   <arg name="car_name" value="$(arg car1_name)"/>
   <arg name="racecar_version" value="racecar-uw-nano"/>
   <arg name="racecar_color" value=""/>
  </include>
  <node name="controller" pkg="nhttc_ros" type="nhttc_ros_node">
   <param name="car_name" value="car1"/>
  </node>
 </group>
 <!-- Launch second car -->
 <group ns="$(arg car2_name)">
  <include file="$(find mushr_sim)/launch/single_car.launch">
   <arg name="car_name" value="$(arg car2_name)"/>
   <arg name="racecar_version" value="racecar-uw-nano"/>
   <arg name="racecar_color" value="-green-black"/>
  </include>
  <node name="controller" pkg="nhttc_ros" type="nhttc_ros_node">
   <param name="car_name" value="car2"/>
  </node>
 </group>
 <group ns="$(arg car3_name)">
  <include file="$(find mushr_sim)/launch/single_car.launch">
   <arg name="car_name" value="$(arg car3_name)"/>
   <arg name="racecar_version" value="racecar-uw-nano"/>
   <arg name="racecar_color" value="-red-white"/>
  </include>
  <node name="controller3" pkg="nhttc_ros" type="nhttc_ros_node">
   <param name="car_name" value="car3"/>
  </node>
 </group>
 <group ns="$(arg car4_name)">
  <include file="$(find mushr_sim)/launch/single_car.launch">
   <arg name="car_name" value="$(arg car4_name)"/>
   <arg name="racecar_version" value="racecar-uw-nano"/>
   <arg name="racecar_color" value="-purple-gold"/>
  </include>
  <node name="controller4" pkg="nhttc_ros" type="nhttc_ros_node">
   <param name="car_name" value="car4"/>
  </node>
 </group>
 <node name="initializer" pkg="nhttc_ros" type="nhttc_pose_init"/>
 <node args="-q $(find nhttc_ros)/bags/$(arg bag_name).bag" name="rosbag_play" output="screen" pkg="rosbag" type="play"/>
</launch>